Foundation/URLCache.swift

```diff
@@ -142,13 +147,37 @@ open class URLCache : NSObject {
     */
     open class var shared: URLCache {
         get {
-            NSUnimplemented()
+            return sharedSyncQ.sync {
+                if let cache = sharedCache {
+                    return cache
+                } else {
+                    let fourMegaByte = 4 * 1024 * 1024
+                    let twentyMegaByte = 20 * 1024 * 1024
+                    let bundleIdentifier = Bundle.main.bundleIdentifier ?? UUID().uuidString
+                    let cacheDirectoryPath = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first?.path ?? "\(NSHomeDirectory())/Library/Caches"
+                    let diskPath = cacheDirectoryPath + "/" + bundleIdentifier
+                    let cache = URLCache(memoryCapacity: fourMegaByte, diskCapacity: twentyMegaByte, diskPath: diskPath)
+                    sharedCache = cache
+                    cache.persistence?.setupCacheDirectory()
+                    return cache
+                }
+            }
         }
         set {
-            NSUnimplemented()
+            guard newValue !== sharedCache else { return }
+            
+            sharedSyncQ.sync {
+                // Remove previous data before resetting new URLCache instance
+                URLCache.sharedCache?.persistence?.deleteAllCaches()
```

Comment:
Does the original implementation do this? Feels weird removing “user” data (even if it is a cache), just by assigning another value.

Reply:
It does not.